doc/policy/version3_transactions.md

```diff
@@ -0,0 +1,100 @@
+difficult to predict, mempool policy restrictions may accidentally allow a malicious party to pin a
+transaction. For example, imagine that counterparties Alice (honest) and Mallory (malicious) have
+conflicting transactions A and B, respectively. Mallory may be able to prevent A from being accepted
+by first submitting B and taking advantage of some limitations and heuristics used in [RBF
+policy](./mempool-replacements.md).
+
+### "Rule 3" Pinning
+
+RBF rules require the replacement transaction pay a higher absolute fee than the aggregate fees paid
+by all original transactions. This means Mallory may increase the fees required to replace B by:
+
+1. Adding transaction(s) that descend from B and pay a feerate too low to fee-bump B through CPFP.
+   For example, assuming the default descendant size limit is 101KvB and B is 1000vB paying a
+feerate of 2sat/vB, adding a 100KvB, 2sat/vB child increases the cost to replace B by 200Ksat.
+
+2. Adding a high-fee descendant of B that also spends from a large, low-feerate mempool transaction,
+   C. The child may pay a very large fee but not actually be fee-bumping B if its overall ancestor
+feerate is still lower than B's individual feerate. For example, assuming the default ancestor size
+limit is 101KvB, B is 1000vB paying 2sat/vB, and C is 99KvB paying 1sat/vB, adding a 1000vB child of
+B and C increases the cost to replace B by 101Ksat.
+
+## Version 3 Rules
+
+All existing standardness rules apply to V3. The following set of additional restrictions apply to
+V3 transactions:
+
+1. A V3 transaction can be replaced, even if it does not signal BIP125 replaceability. Other
+   conditions apply, see [RBF rules](./mempool-replacements.md) and [Package RBF
+rules][./packages.md#Package-Replace-By-Fee].
+
+2. Any descendant of an unconfirmed V3 transaction must also be V3.
+
+*Rationale*: Combined with Rule 1, this gives us the property of "inherited signaling" when
+descendants of unconfirmed transactions are created. Additionally, checking whether a transaction
+signals replaceability this way does not require mempool traversal, and does not change based on
+what transactions are mined.
+
+*Note*: A V3 transaction can spend outputs from *confirmed* non-V3 transactions.
+
+3. A V3 transaction's unconfirmed ancestors must all be V3.
+
+*Rationale*: Ensure the ancestor feerate rule does not underestimate a to-be-replaced V3 mempool
+transaction's incentive compatibility. Imagine the original transaction, A, has a child B and
+co-parent C (i.e. B spends from A and C). C also has another child, D. B is one of the original
+transactions and thus its ancestor feerate must be lower than the package's. However, this may be an
+underestimation because D can bump C without B's help. This is resolved if V3 transactions can only
+have V3 ancestors, as then C cannot have another child.
+
+4. A V3 transaction cannot have more than 1 unconfirmed descendant.
```

Comment:
Question about this and sorry if it's already been addressed:

Does this require extra logic for reorg handling? I'm thinking of a chain of v3 TXs:

tx1 (confirmed in block 100) ->
tx2 (confirmed in block 101) ->
tx3 (unconfirmed, currently in mempool)

if block 101 is disconnected, that potentially puts tx2 back in the mempool meaning tx3 is now invalid and should be evicted.

Reply:
Yes, v3 enforcement during reorg is implemented in `MaybeUpdateMempoolForReorg` [here](https://github.com/bitcoin/bitcoin/pull/25038/files#diff-97c3a52bc5fad452d82670a7fd291800bae20c7bc35bb82686c2c0a4ea7b5b98R364-R370) and tested in mempool_accept_v3.py [here](https://github.com/bitcoin/bitcoin/pull/25038/files#diff-15a1888c9151fc1d182c23e34b71d691f70df448bceb9eb78c8296f18854b6a3R163). Would it make sense for me to add a note in the docs?

Reply:
Sweet 👍 Couldn't hurt to mention that in a bullet point here I suppose.